### PR TITLE
feat(shield): fetch service data api

### PR DESCRIPTION
**Pull request by FemiNoviaLina**
Add fetch service data api for user and group entity.

### CODE REVIEW

**Comment by FemiNoviaLina** [comment]
will move from admin when [upsert PR](https://github.com/goto/proton/pull/99) merged.

File: gotocompany/shield/v1beta1/shield.proto [hunk 1]
```diff
@@ -504,6 +504,26 @@ service ShieldService {
       summary: "Create Service Data Key";
     };
   }
+
+  rpc GetUserServiceData(GetUserServiceDataRequest) returns (GetUserServiceDataResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/users/{id}/servicedata"
```
Comment (mabdh) [inline gotocompany/shield/v1beta1/shield.proto:510]:
```suggestion
      get: "/v1beta1/users/{user_id}/servicedata"
```

File: gotocompany/shield/v1beta1/shield.proto [hunk 2]
```diff
@@ -504,6 +504,26 @@ service ShieldService {
...
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Service Data";
+      summary: "Get User Service Data Key";
+    };
+  }
+
+  rpc GetGroupServiceData(GetGroupServiceDataRequest) returns (GetGroupServiceDataResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/groups/{id}/servicedata"
```
Comment (mabdh) [inline gotocompany/shield/v1beta1/shield.proto:520]:
```suggestion
      get: "/v1beta1/groups/{group_id}/servicedata"
```

File: gotocompany/shield/v1beta1/shield.proto [hunk 3]
```diff
@@ -1168,4 +1188,24 @@ message CreateServiceDataKeyRequest {
...
+
+message GetUserServiceDataRequest {
+  string id = 1;
+  repeated string entity = 2 [(validate.rules).repeated.items.string = {in: "user", in: "group"}];
+  string project = 3;
+}
+
+message GetGroupServiceDataRequest {
+  string id = 1;
+  repeated string entity = 2 [(validate.rules).repeated.items.string = {in: "user", in: "group"}];
```
Comment (mabdh) [inline gotocompany/shield/v1beta1/shield.proto:1201]: we don't need to have this for group right?
```suggestion
```

File: gotocompany/shield/v1beta1/servicedata.proto [hunk 4]
```diff
@@ -94,3 +116,22 @@ message UpsertUserServiceDataResponse {
...
+  string project = 3;
+}
+
+message GetGroupServiceDataRequest {
+  string group_id = 1;
+  string project = 2;
+}
+
+message GetUserServiceDataResponse {
+  google.protobuf.Struct servicedata = 1;
```
Comment (mabdh) [inline gotocompany/shield/v1beta1/servicedata.proto:132]: or probably we can just call this `data` ?

**Review by mabdh — APPROVED** [review]

**Pull request merged**